grid/grid_network.py

```diff
@@ -0,0 +1,50 @@
+import requests
+import json
+import syft as sy
+from grid.websocket_client import WebsocketGridClient
+
+
+class GridNetwork(object):
+    """  The purpose of the Grid Network class is to control the entire communication flow by abstracting operational steps.
+    
+        Attributes:
+            - gateway_url : network address to which you want to connect
+            - connected_grid_nodes : Grid nodes that are connected to the application.
+    """
+
+    def __init__(self, gateway_url):
+        self.gateway_url = gateway_url
+        self.connected_grid_nodes = {}
+
+    def search(self, *query):
+        """ Search a set of tags across the grid network 
+            
+            Parameters:
+                query : A set of dataset tags
+            Return:
+                tensor_set : matrix of tensor pointers
+        """
+        body = json.dumps({"query": list(query)})
+
+        # Asks to grid gateway about dataset-tags
+        response = requests.post(self.gateway_url + "/search", data=body)
+
+        # List of nodes that contains the desired dataset
+        match_nodes = json.loads(response.content)
+
+        # Connect with grid nodes that contains the dataset and get their pointers
+        tensor_set = []
+        for node in match_nodes:
```

Comment:
instead of using node[0] and node[1] can you use more descriptive names?

You can interate in `match_nodes` using a tuple: `for x, y in match_nodes`

Reply:
Done! (7b5588b)